packages/contractkit/src/providers/celo-provider.ts

```diff
@@ -131,6 +131,47 @@ export class CeloProvider {
     }
   }
 
+  on(type: string, callback: () => void): void {
```

Comment:
i *think* that web3.js check `provider has "on"` to check if the provider supports subscriptions.

So, i would define these properties only if the existing provider supports them